src/tdycore.c

```diff
@@ -197,6 +197,18 @@ PetscErrorCode TDySetDM(TDy tdy, DM dm) {
   PetscFunctionReturn(0);
 }
 
+PetscReal TDyGetGravity() {
+  PetscErrorCode ierr;
+  PetscReal gravity = 9.8068;
+
+  ierr = PetscOptionsBegin(PETSC_COMM_WORLD,NULL,"General Options",""); CHKERRQ(ierr);
+  ierr = PetscOptionsReal("-tdy_gravity", "Value of gravity", NULL, gravity, &gravity, NULL); CHKERRQ(ierr);
+  ierr = PetscOptionsEnd(); CHKERRQ(ierr);
```

Comment:
This is okay, but probably better to collect all constants in a big struct somewhere so the options only need to get parsed once. One other thing I do sometimes is abstract over units, so I write
```c
PetscReal gravity = 9.8068 * meter / (second * second);
```
where `meter` is the size of one meter in internal length units and `second` is the duration of one second in internal time units. Being able to choose characteristic scales like this is useful to scale systems for solvers and time integrator tolerances, and allows one to check for dimensional consistency. In many cases, all user input and output should be converted back to SI units (loses bit-for-bit if you do this with checkpoints), or store the scale factors in the output files so visualization and other human interfaces always see SI.
https://github.com/CEED/libCEED/blob/main/examples/fluids/navierstokes.c#L243